qiskit/transpiler/passes/utils/error.py

```diff
@@ -0,0 +1,58 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Error pass to be called when an error happens."""
+
+import logging
+import string
+import warnings
+
+from qiskit.transpiler.basepasses import AnalysisPass
+from qiskit.transpiler.exceptions import TranspilerError
+
+
+class Error(AnalysisPass):
+    """Error pass to be called when an error happens."""
+
+    def __init__(self, msg=None, action='raise'):
+        """Error pass.
+
+        Args:
+            msg (str): Error message, if not provided a generic error will be used
+            action (str): the action to perform. Default: 'raise'. The options are:
+              * 'raise': Raises a `TranspilerError` exception with msg
+              * 'warning': Raises a non-fatal warning with msg
+              * 'logger': logs in `logging.getLogger(__name__)`
+              * other: logs in `logging.getLogger(action)`
```

Comment:
Is there a need to accept an arbitrary logging target? Maybe it can be an optional kwarg?

Reply:
I removed this option in b588e94bb235d423a0954a48942fdd25c89ed080 We can add the optional kwarg if needed in the future